src/adapter/src/coord/cluster_scheduling.rs

```diff
@@ -0,0 +1,251 @@
+                }
+                Some(managed_config) => {
+                    if matches!(managed_config.schedule, ClusterScheduleOptionValue::Manual) {
+                        debug!(
+                            "handle_scheduling_decisions: \
+                            Removing cluster {} from cluster_scheduling_decisions, \
+                            because schedule is Manual",
+                            cluster_id
+                        );
+                        self.cluster_scheduling_decisions.remove(&cluster_id);
+                    }
+                }
+            }
+        }
+
+        // 3. Act on `scheduling_decisions` where needed.
+        let mut altered_a_cluster = false;
+        for (cluster_id, decisions) in self.cluster_scheduling_decisions.clone() {
+            // If all policies have made a decision about this cluster
+            if POLICIES.iter().all(|policy| decisions.contains_key(policy)) {
+                // check whether the cluster's state matches the needed state.
+                let needs_replica = decisions.values().contains(&true);
+                let cluster_config = self
+                    .get_managed_cluster_config(cluster_id)
+                    .expect("cleaned up non-existing and unmanaged clusters above");
+                let has_replica = cluster_config.replication_factor > 0; // Is it On?
+                if needs_replica != has_replica {
+                    // Turn the cluster On or Off.
+                    altered_a_cluster = true;
+                    let mut new_config = cluster_config.clone();
+                    new_config.replication_factor = if needs_replica { 1 } else { 0 };
+                    if let Err(e) = self
+                        .sequence_alter_cluster_managed_to_managed(
+                            None,
+                            cluster_id,
+                            &cluster_config,
+                            new_config.clone(),
+                        )
+                        .await
+                    {
+                        soft_panic_or_log!(
+                            "handle_scheduling_decisions couldn't alter cluster {}. \
+                             Old config: {:?}, \
+                             New config: {:?}, \
+                             Error: {}",
+                            cluster_id,
+                            cluster_config,
+                            new_config,
+                            e
+                        );
+                    }
+                }
+            } else {
+                debug!(
+                    "handle_scheduling_decisions: \
+                    Not all policies have made a decision about cluster {}. decisions: {:?}",
+                    cluster_id, decisions,
+                );
+            }
+        }
+
+        self.metrics
+            .handle_scheduling_decisions_seconds
+            .with_label_values(&[altered_a_cluster.to_string().as_str()])
```

Comment:
This will cause the metric to have unbounded cardinality, which is not good! If we decide to keep these metrics (which I don't think we need to, see below comment) can you make sure to remove this label?

Reply:
`altered_a_cluster` is a boolean, so this just doubles the cardinality. Is that ok? E.g.:
```
# HELP mz_handle_scheduling_decisions_seconds The time `handle_scheduling_decisions` takes.
# TYPE mz_handle_scheduling_decisions_seconds histogram
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.000128"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.000256"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.000512"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.001"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.002"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.004"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.008"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.016"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.032"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.064"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.128"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.256"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="0.512"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="1"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="2"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="4"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="8"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="false",le="+Inf"} 225
mz_handle_scheduling_decisions_seconds_sum{altered_a_cluster="false"} 0.0035463509999999953
mz_handle_scheduling_decisions_seconds_count{altered_a_cluster="false"} 225
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.000128"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.000256"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.000512"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.001"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.002"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.004"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.008"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.016"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.032"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.064"} 0
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.128"} 2
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.256"} 69
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="0.512"} 94
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="1"} 164
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="2"} 277
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="4"} 277
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="8"} 277
mz_handle_scheduling_decisions_seconds_bucket{altered_a_cluster="true",le="+Inf"} 277
mz_handle_scheduling_decisions_seconds_sum{altered_a_cluster="true"} 207.61221440799994
mz_handle_scheduling_decisions_seconds_count{altered_a_cluster="true"} 277
```

Reply:
I see, I misunderstood the `altered_a_cluster` label, this is fine!